students/shibata/session3/strings.py

```diff
@@ -0,0 +1,13 @@
+"""
+you want to find a string formatting operator that will “pad” the number with zeros for you.
+
+( 2, 123.4567, 10000, 12345.67)
+
+and produce:
+
+'file_002 :   123.46, 1.00e+04, 1.23e+04'
+"""
+
+array = (2, 123.4567, 10000, 12345.67)
+print ("file_00",array[0],":","{:.2f}".format(array[1]),"{:.2e}".format(array[2]),"{:.2e}".format(array[3]))
```

Comment:
the goal here was to use the formatting language to pad those zeros. The way you have done it works for single digits: 002, but would fail for two digits (say the tenth file) -- you want it to give you 010, not 0010.

There is a way to use the formatting language operator to pad a value with however many zeros you need to make it a total width of 3 (or any size you want).

Also, you can put multiple formatters in one string, and then pass multiple values in -- so this would be something like:

``"file_{} : {:.2f},  {:.2e}, {:.3g}".format(2, 123.4567, 10000, 12345.67)``

I'll leave it to you to figure out what to put in that first bracket :-)

And note the "g" formatter, slightly different than the "f" -- it does significant digits, rather than number of decimal places.